pkg/cluster/comparison.go

```diff
@@ -0,0 +1,159 @@
+package cluster
+
+import (
+	"reflect"
+
+	u "github.com/zalando-incubator/postgres-operator/pkg/util"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/apps/v1beta1"
+)
+
+type ContainerCondition func(a, b *v1.Container) bool
+type StatefulSetCondition func(a, b *v1beta1.StatefulSet) bool
+type VolumeClaimCondition func(a, b *v1.PersistentVolumeClaim) bool
+
+type ResourceCheck struct {
+	containerCondition   ContainerCondition
+	statefulSetCondition StatefulSetCondition
+	volumeClaimCondition VolumeClaimCondition
+	result               Result
+	reason               string
+}
+
+type Result struct {
+	needUpdate      *bool
```

Comment:
why do we need pointers here? We could just default to false. This should also eliminate ugly True, False and isTrue